controller/features.go

```diff
@@ -2,29 +2,103 @@ package controller
 	uuid "github.com/satori/go.uuid"
+	"strings"
 )
 
 // FeaturesController implements the features resource.
 type FeaturesController struct {
 	*goa.Controller
+	config *configuration.Data
 }
 
 // NewFeaturesController creates a features controller.
-func NewFeaturesController(service *goa.Service) *FeaturesController {
-	return &FeaturesController{Controller: service.NewController("FeaturesController")}
+func NewFeaturesController(service *goa.Service, config *configuration.Data) *FeaturesController {
+	return &FeaturesController{
+		Controller: service.NewController("FeaturesController"),
+		config:     config,
+	}
 }
 
 // List runs the list action.
 func (c *FeaturesController) List(ctx *app.ListFeaturesContext) error {
 	// FeaturesController_List: start_implement
 
-	// Put your logic here
+	jwtToken := goajwt.ContextJWT(ctx)
+	groupId := ""
+	if jwtToken != nil {
+		groupId = jwtToken.Claims.(jwtgo.MapClaims)["company"].(string) // TODO replace with a custom claim group_id
+	}
+	var enableFeatures *app.FeatureList
+	if c.config != nil {
+		featuresListFromUnleash, err := getFeatureListFromUnleashServer(ctx, c.config, groupId)
+		if err != nil {
+			log.Error(ctx.Context, map[string]interface{}{
+				"err": err,
+			}, "Unable to connect to Unleash server")
+			return ctx.Err()
+		}
+		log.Debug(ctx, nil, "FEATURES: %s", featuresListFromUnleash)
+		enableFeatures = featuresListFromUnleash
+	} else { // TEST Data
+		enabledFeaturesList := app.FeatureList{Data: []*app.Feature{}}
+		featuresList := buildFeaturesList(5)
+		for _, elt := range featuresList.Data {
+			if elt.Attributes.GroupID != nil && strings.ToLower(*elt.Attributes.GroupID) == strings.ToLower(groupId) {
+				enabledFeaturesList.Data = append(enabledFeaturesList.Data, elt)
+			}
+		}
+		enableFeatures = &enabledFeaturesList
+	}
 
 	// FeaturesController_List: end_implement
-	return ctx.OK(buildFeaturesList(5))
+	return ctx.OK(enableFeatures)
+}
+
+func getFeatureListFromUnleashServer(ctx *app.ListFeaturesContext, config *configuration.Data, groupId string) (*app.FeatureList, error) {
+	res := app.FeatureList{}
+	toggleClient, err := featuretoggles.NewFeatureToggleClient(ctx.Context, config)
+	if err != nil {
+		log.Error(ctx.Context, map[string]interface{}{
+			"addr": config.GetHTTPAddress(),
+			"err":  err,
+		}, "Unable to connect to Unleash server")
+		return nil, err
+	}
+
+	listOfFeatures := toggleClient.GetEnabledFeatures(groupId)
+	res = format(listOfFeatures, groupId)
+	return &res, nil
+}
+
+func format(list []string, groupId string) app.FeatureList {
+	res := app.FeatureList{}
+	for i := 0; i < len(list); i++ {
+		// TODO remove ID, make unleash client return description
+		ID := uuid.NewV4()
+		descriptionFeature := "Description of the feature"
+		enabledFeature := true
+		nameFeature := list[i]
+
+		feature := app.Feature{
+			ID: ID,
+			Attributes: &app.FeatureAttributes{
+				Description: &descriptionFeature,
+				Enabled:     &enabledFeature,
```

Comment:
why would you need that `Enabled` attribute ?

Reply:
@xcoulon we should change: to return a list of struct bc atm we return only a list of string (with feature name) see: https://github.com/xcoulon/unleash-client-go/commit/377eea70f16806a6f68414ceed39b1f750eafe1f#diff-d0b7f86cea13f90d3cc0cc2338da8d5fR266